### ISSUE TITLE
Screensharing Broken (xdg-desktop-portal-hyprland-1.3.7-1.1)

### ISSUE THREAD

**Issue by Ly-sec** [hyprwm/xdg-desktop-portal-hyprland#280]
Hello,
it seems like the last update (xdg-desktop-portal-hyprland-1.3.7-1.1) broke screensharing to some degree for discord(vesktop etc).
You're still able to share the whole screen, however sharing a single window does not work anymore. You're able to select it once and after the usual discord screenshare popup the window selection is empty.

**Comment by Ly-sec** [comment]
Well, I downgraded to xdg-desktop-portal-hyprland-1.3.6 and it just works.
Not sure what happened there but I'll wait until that issue is solved.

**Comment by herbiejhopkins** [comment]
Just hit this same issue. Seems the first request to screencopy a window succeeds but the subsequent ones just shows no windows to select
![image](https://github.com/user-attachments/assets/56c9060e-0ea4-4bca-9463-5138e4d42828)
Some apps seems to make two requests when attempting to screen share which always fails since only the first of these succeeds.

**Comment by Ly-sec** [comment]
> Just hit this same issue. Seems the first request to screencopy a window succeeds but the subsequent ones just shows no windows to select ![image](https://private-user-images.githubusercontent.com/56479007/380224976-56c9060e-0ea4-4bca-9463-5138e4d42828.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4Njk5OTcsIm5iZiI6MTcyOTg2OTY5NywicGF0aCI6Ii81NjQ3OTAwNy8zODAyMjQ5NzYtNTZjOTA2MGUtMGVhNC00YmNhLTk0NjMtNTEzOGU0ZDQyODI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDE1MjEzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzODI5ZTIxY2M2Y2JhZDI0N2Y3Y2M3MzQ4YjI4YjQzM2NlMzZhNzAyNmFhYjlkNGUxZjIzZWI2YTVjZDQyZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q5Ik1OW0MT6EpeGYEXh9WpTGXGEqiFt9bOZmcV_bcu0) Some apps seems to make two requests when attempting to screen share which always fails since only the first of these succeeds.

Yup, I almost think it has something to do with `sdbus-cpp`, since that seems to be the only big change that I could see in the last update

**Comment by s0mebodyhelpme** [comment]
experiencing the same issue on vesktop. yeah it seems to have been introduced recently.

**Comment by Alextibtab** [comment]
Also been having a similar issue trying to screenshare windows through vesktop. 

Initial portal prompt where i can see windows fine
![image](https://github.com/user-attachments/assets/dd02a8d2-e8e1-49f3-af01-7eb2794575c1)

Then I get vesktop preview and once I click go live I get reprompted where the windows are now blank
![image](https://github.com/user-attachments/assets/665194a9-4fbb-4ed5-8509-fb9c26a8895a)
![image](https://github.com/user-attachments/assets/3ad68aad-b5ca-46f2-8f92-0922bbe9d424)

if I run `systemctl --user restart xdg-desktop-portal-hyprland` while this window is open I can then see the windows again and screensharing works

The initial portal prompt has the correct values in the XDPH_WINDOW_SHARING_LIST
`XDPH_WINDOW_SHARING_LIST='2254054864[HC>]firefox[HT>]Screensharing Broken (xdg-de etc......`

however after that initial one it will always then be 
`XDPH_WINDOW_SHARING_LIST=''`

until I restart the service

**Comment by Eveeifyeve** [comment]
CC @vaxerski, this is a current issue with xdph 1.3.7 seems like it could be sbus implementation issue?

**Comment by Alextibtab** [comment]
Here's the XDPH -v output if it's of use to anyone in it I click screenshare on vesktop the first portal prompt has the `XDPH_WINDOW_SHARING_LIST` correctly set but then anything after that it's always just empty

[xdgh-portal-bug.txt](https://github.com/user-attachments/files/17528859/xdgh-portal-bug.txt)

**Comment by Eveeifyeve** [comment]
The only fix I could think for now is either wait for a fix (CC @vaxerski PR: #278 causing issues potentially at line 35-36 `ScreenCopy.cpp`, downgrade to 1.3.6 or screen share full-screen (not individual windows)

**Comment by Oery** [comment]
Not sure if it's related to the same issue but I lost Screen Capture in OBS. In Vesktop I can share my screen but in OBS the source just isn't in the menu anymore. Also can confirm the issue with windows.

**Comment by Alextibtab** [comment]
After cloning and building 1.3.6 with sdbus 1.6.0 I can confirm that screensharing worked perfectly for me and after adding some additional logging to the 1.3.7 repo the `buildWindowList()` function in ScreencopyShared.cpp correctly get's the top level windows on the initial call but then any additional calls the portalmanagers toplevels are always empty.

```ps1
[LOG] Building window list...
[LOG] Found 5 toplevel windows
[LOG] Adding window: handle=665542048, class='spotify', title='Spotify Premium'
[LOG] Adding window: handle=665543376, class='firefox', title='core: move to sdbus-cpp2 (#278) · 5c72a7fea1 - hyprwm/xdg-desktop-portal-hyprland - Hyprland Development — Mozilla Firefox'
[LOG] Adding window: handle=665544816, class='Alacritty', title='~/projects/c++/xdph/xdg-desktop-portal-hyprland-1-3-7> nvim'
[LOG] Adding window: handle=665547648, class='vesktop', title='(160) Discord | Lobby | RIP Ye Mighty Vessel'
[LOG] Adding window: handle=665549008, class='Alacritty', title='~/projects/c++/xdph/xdg-desktop-portal-hyprland-1-3-7> ./xdg-desktop-portal-hyprland'
[LOG] execAndGet: WAYLAND_DISPLAY='wayland-1' QT_QPA_PLATFORM='wayland' XCURSOR_SIZE='24' HYPRLAND_INSTANCE_SIGNATURE='4520b30d498daca8079365bdb909a8dea38e8d55_1729949002_654540548' XDPH_WINDOW_SHARING_LIST='665542048[HC>]spotify[HT>]Spotify Premium[HE>]665543376[HC>]firefox[HT>]core: move to sdbus-cpp2 (#278) · 5c72a7fea1 - hyprwm/xdg-desktop-portal-hyprland - Hyprland Development — Mozilla Firefox[HE>]665544816[HC>]Alacritty[HT>]~/projects/c++/xdph/xdg-desktop-portal-hyprland-1-3-7> nvim[HE>]665546176[HC>]vesktop[HT>](160) Discord | Lobby | RIP Ye Mighty Vessel[HE>]665549008[HC>]Alacritty[HT>]~/projects/c++/xdph/xdg-desktop-portal-hyprland-1-3-7> ./xdg-desktop-portal-hyprland[HE>]' hyprland-share-picker 2>&1
[LOG] [sc] Selection: /window:665542048
```

```ps1
[LOG] Building window list...
[LOG] Found 0 toplevel windows
[LOG] execAndGet: WAYLAND_DISPLAY='wayland-1' QT_QPA_PLATFORM='wayland' XCURSOR_SIZE='24' HYPRLAND_INSTANCE_SIGNATURE='4520b30d498daca8079365bdb909a8dea38e8d55_1729949002_654540548' XDPH_WINDOW_SHARING_LIST='' hyprland-share-picker 2>&1
```

**Comment by Alextibtab** [comment]
Created [PR](https://github.com/hyprwm/xdg-desktop-portal-hyprland/pull/283) to fix the issue
